### PR TITLE
Fixes for DF cloud section

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes the following:
- don't display "You will incur a charge from your cloud provider." when local
- the choose cloud title should change depending on the cloud chosen
- clicking "Change cloud" should clear the chosen credential

### CODE REVIEW

**Comment by huwshimi** [comment]
QA: not currently able to QA the DF. Tests should pass.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5580/
Test PASSed.

File: .gitignore [hunk 1]
```diff
@@ -42,5 +42,6 @@ uitest/
 upload_release.py
 .vagrant
 virtualenv
+xvfb_pid
```
Comment (huwshimi) [inline .gitignore]: This file will exist if you leave `start-karma` running.

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 2]
```diff
@@ -220,6 +220,17 @@ YUI.add('deployment-flow', function() {
     },
 
     /**
+      Handle clearing the chosen cloud.
+
+      @method _changeCloud
+    */
+    _changeCloud: function() {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: As this function only clears the selected cloud. Maybe it should be named as such. For example `_clearCloud`.

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 3]
```diff
@@ -256,6 +267,23 @@ YUI.add('deployment-flow', function() {
...
+      @returns {String} The cloud title.
+    */
+    _generateCloudTitle: function() {
+      var cloud = this.state.cloud;
+      if (!cloud) {
+        return 'Choose cloud to deploy to';
+      } else if (cloud === 'local') {
+        return 'Local cloud';
+      } else {
+        return 'Public cloud';
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Do we also need to handle private clouds?

**Comment by anthonydillon** [comment]
Thanks @huwshimi, looks good with a few questions :+1:

File: jujugui/static/gui/src/app/components/deployment-flow/machines/machines.js [hunk 4]
```diff
@@ -71,12 +71,16 @@ YUI.add('deployment-machines', function() {
     },
 
     render: function() {
+      var chargeMessage;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/machines/machines.js]: I think that this should have a default value of `''`

**Comment by hatched** [comment]
Thanks @huwshimi I think this can land with the trivial comments addressed.

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: I don't think we have any way to work with them yet.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5588/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @anthonydillon, @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**